pkg/ebpf/bytecode/runtime/printk_patcher_test.go

```diff
@@ -45,7 +45,8 @@ func TestPatchPrintkNewline(t *testing.T) {
 	if kernelVersion.Code <= ebpfkernel.Kernel5_9 {
 		t.Skip("Skipping test on older kernels, instruction patching not used there")
 	}
-	if kernelVersion.IsDebianKernel() && kernelVersion.Code <= ebpfkernel.Kernel5_10 {
+	if (kernelVersion.IsDebianKernel() && kernelVersion.Code <= ebpfkernel.Kernel5_11) ||
+		(kernelVersion.IsAmazonLinuxKernel() && kernelVersion.Code <= ebpfkernel.Kernel5_10) {
 		t.Skip("Tracing not available, cannot test")
```

Comment:
By tracing, do you mean the `trace_pipe`/`trace buffer`? If so, then that should be available on all these kernels.

Reply:
Maybe it is mounted at a different location. The ftrace control files can be mounted at debugfs, i.e. `/sys/kernel/debug/tracing/`, or `/sys/kernel/trace/`.

Reply:
It is technically available, but I've found that in some cases it isn't enabled or we cannot read from it (or do so inconsistently). In these tests I am getting a timeout because nothing is coming through. I'll check if it's in a different file.

Reply:
Maybe instead of `trace_pipe` you can try reading the `trace` file.
The difference is explain in this document: https://www.kernel.org/doc/Documentation/trace/ftrace.txt